### PR TITLE
feat: Adds samples for foreign key on delete cascade action.

### PR DESCRIPTION
**Pull request by anuragsrivstv**

### CODE REVIEW

**Comment by anuragsrivstv** [comment]
Currently samples are tested in the staging environment. Will test it again once the feature is available and sample integration tests are successful with it.  Keeping this PR as draft till then.

**Comment by anuragsrivstv** [comment]
> Currently samples are tested in the staging environment. Will test it again once the feature is available and sample integration tests are successful with it. Keeping this PR as draft till then.

Now this PR can be reviewed.

File: spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            Assert.Contains("ShoppingCarts", tables);
+        });
+    }
+
+    private async Task<List<string>> ListTableNamesAsync(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+        using var connection = new SpannerConnection(connectionString);
+
+        var command = connection.CreateSelectCommand("SELECT table_name FROM information_schema.tables WHERE table_schema='public' OR table_schema=''");
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs:49]:
```suggestion
        var command = connection.CreateSelectCommand("SELECT Count(*) FROM information_schema.tables WHERE table_name='Customers' OR table_name='ShoppingCarts'");
```

File: spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        var tableNames = new List<string>();
+        using var reader = await command.ExecuteReaderAsync();
+        while (await reader.ReadAsync())
+        {
+            tableNames.Add(reader.GetFieldValue<string>("table_name"));
+        }
+
+        return tableNames;
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs:57]:
```suggestion
        var tableCount = await command.ExecuteScalarAsync();
        Assert.Equals(2, tableCount);
```

File: spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            var sample = new CreateTableWithForeignKeyDeleteCascadeAsyncSample();
+            await sample.CreateTableWithForeignKeyDeleteCascadeAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            var tables = await ListTableNamesAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Contains("Customers", tables);
+            Assert.Contains("ShoppingCarts", tables);
+        });
+    }
+
+    private async Task<List<string>> ListTableNamesAsync(string projectId, string instanceId, string databaseId)
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs:44]:
```suggestion
    private async Task AssertTablesCreatedAsync(string projectId, string instanceId, string databaseId)
```

File: spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    public async Task CreateTableWithForeignKeyDeleteCascadeAsync()
+    {
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            var sample = new CreateTableWithForeignKeyDeleteCascadeAsyncSample();
+            await sample.CreateTableWithForeignKeyDeleteCascadeAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            var tables = await ListTableNamesAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.Contains("Customers", tables);
+            Assert.Contains("ShoppingCarts", tables);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateTableWithForeignKeyDeleteCascadeAsyncTest.cs:40]:
```suggestion
            await AssertTablesCreatedAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
```

File: spanner/api/Spanner.Samples.Tests/AlterTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            CartId INT64 NOT NULL, 
+            CustomerId INT64 NOT NULL, 
+            CustomerName STRING(62) NOT NULL
+            ) PRIMARY KEY (CartId)";
+
+        var cmd = connection.CreateDdlCommand(createCustomerTableStatement, createShoppingCartTableStatement);
+        await cmd.ExecuteNonQueryAsync();
+    }
+
+    private async Task<List<string>> ListDeleteRuleConstraintsAsync(string projectId, string instanceId, string databaseId)
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/AlterTableWithForeignKeyDeleteCascadeAsyncTest.cs:66]: Change this one similar to the others.

File: spanner/api/Spanner.Samples.Tests/AlterTableWithForeignKeyDeleteCascadeAsyncTest.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        var connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+        using var connection = new SpannerConnection(connectionString);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/AlterTableWithForeignKeyDeleteCascadeAsyncTest.cs:69]: Reuse the connection from the table creation. You can create the connection on the main method and pass it as a parameter to both helper methods.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
A few suggestions that I think will improve tests.

**Review by amanda-tarafa — APPROVED** [review]
Thanks!

**Pull request merged**